saga/seata-saga-engine/src/main/java/io/seata/saga/engine/expression/impl/DefaultExpressionResolver.java

```diff
@@ -0,0 +1,138 @@
+import io.seata.saga.engine.expression.ExpressionFactory;
+import io.seata.saga.engine.expression.ExpressionFactoryManager;
+import io.seata.saga.engine.expression.ExpressionResolver;
+import io.seata.saga.engine.expression.exception.ExceptionMatchExpression;
+import io.seata.saga.engine.expression.spel.SpringELExpression;
+import io.seata.saga.proctrl.ProcessContext;
+import io.seata.saga.statelang.domain.DomainConstants;
+
+/**
+ * Default {@link ExpressionResolver} implementation
+ *
+ * @author ptyin
+ */
+public class DefaultExpressionResolver implements ExpressionResolver {
+
+    protected static class ExpressionStruct {
+        int typeStart;
+
+        int typeEnd;
+
+        int end;
+
+        String type;
+
+        String content;
+    }
+
+    private ExpressionFactoryManager expressionFactoryManager;
+
+    private String rootObjectName;
+
+    @Override
+    public Expression getExpression(String expressionStr) {
+        ExpressionStruct struct = parseExpressionStruct(expressionStr);
+
+        ExpressionFactory expressionFactory = expressionFactoryManager.getExpressionFactory(struct.type);
+        if (expressionFactory == null) {
+            throw new IllegalArgumentException("Cannot get ExpressionFactory by Type[" + struct + "]");
+        }
+        return expressionFactory.createExpression(struct.content);
+    }
+
+    @Override
+    public <T extends Expression> Object getDefaultElContext(ProcessContext context, Class<T> expressionClass) {
+        if (expressionClass.isAssignableFrom(ExceptionMatchExpression.class)) {
+            return context.getVariable(DomainConstants.VAR_NAME_CURRENT_EXCEPTION);
+        } else if (expressionClass.isAssignableFrom(SpringELExpression.class)) {
+            return context.getVariable(DomainConstants.VAR_NAME_STATEMACHINE_CONTEXT);
+        }
+        return context.getVariables();
+    }
+
+    @Override
+    public <T extends Expression> Object getStatusEvaluationElContext(ProcessContext context, Class<T> expressionClass) {
+        if (expressionClass.isAssignableFrom(ExceptionMatchExpression.class)) {
+            return context.getVariable(DomainConstants.VAR_NAME_CURRENT_EXCEPTION);
+        } else if (expressionClass.isAssignableFrom(SpringELExpression.class)) {
+            return context.getVariable(DomainConstants.VAR_NAME_OUTPUT_PARAMS);
+        }
+        return context.getVariables();
+    }
+
+    protected ExpressionStruct parseExpressionStruct(String expressionStr) {
+        ExpressionStruct struct = new ExpressionStruct();
+        struct.typeStart = expressionStr.indexOf("$");
+        int dot = expressionStr.indexOf(".", struct.typeStart);
+        int leftBracket = expressionStr.indexOf("{", struct.typeStart);
+
+
+        boolean isOldEvaluatorStyle = false;
+        if (struct.typeStart == 0) {
+            if (leftBracket < 0 && dot < 0) {
+                throw new IllegalArgumentException(String.format("Expression [%s] type is not closed", expressionStr));
```

Comment:
if express is '$=1', this message is inappropriate. Whether to modify it to 'resolve expression %s fail ' ?

Reply:
I think the message "resolve expression %s fail" you mentioned is kind of too generic, and may confuse users. And the current message does make sense in case of "$=1". The message would be "Expression [$=1] type is not closed", which can be tranlated as "表达式[$=1]类型未闭合". Since the the standard format of expression is `$expressionType.expressionContent`, this message indicates the `expressionType` is not closed, i.e., the type end indicator '.' is missing.

Reply:
Or, an optimized solution can be "Failed to resolve the expression [%s]: the type end indicator is missing." What do you think?

Reply:
I have priority the second proposal.